src/login/LoginPage.jsx

```diff
@@ -1,435 +1,266 @@
-  getTpaHint,
-  getTpaProvider,
   setSurveyCookie,
   updatePathWithQueryParams,
-  windowScrollTo,
 } from '../data/utils';
 import ResetPasswordSuccess from '../reset-password/ResetPasswordSuccess';
 
-class LoginPage extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-    this.state = {
-      password: this.props.loginFormData.password,
-      emailOrUsername: this.props.loginFormData.emailOrUsername,
-      errors: {
-        emailOrUsername: this.props.loginFormData.errors.emailOrUsername,
-        password: this.props.loginFormData.errors.password,
-      },
-      isSubmitted: false,
-    };
-    this.queryParams = getAllPossibleQueryParams();
-    this.tpaHint = getTpaHint();
-  }
+const LoginPage = (props) => {
+  const {
+    backedUpFormData,
+    loginErrorCode,
+    loginErrorContext,
+    loginResult,
+    shouldBackupState,
+    showResetPasswordSuccessBanner,
+    submitState,
+    // Actions
+    backupFormState,
+  } = props;
+  const { formatMessage } = useIntl();
+  const activationMsgType = getActivationStatus();
+  const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
 
-  componentDidMount() {
+  const [formFields, setFormFields] = useState({ ...backedUpFormData.formFields });
+  const [errorCode, setErrorCode] = useState({ type: '', count: 0 });
+  const [errors, setErrors] = useState({ ...backedUpFormData.errors });
+
+  useEffect(() => {
     sendPageEvent('login_and_registration', 'login');
-    const payload = { ...this.queryParams };
+  }, []);
 
-    if (this.tpaHint) {
-      payload.tpa_hint = this.tpaHint;
+  /**
+   * Backup the login form in redux when login page is toggled.
+   */
+  useEffect(() => {
+    if (shouldBackupState) {
+      backupFormState({
+        formFields: { ...formFields },
+        errors: { ...errors },
+      });
     }
-    this.props.getThirdPartyAuthContext(payload);
-    this.props.loginRequestReset();
-  }
+  }, [shouldBackupState, formFields, errors, backupFormState]);
 
-  shouldComponentUpdate(nextProps) {
-    if (nextProps.loginFormData && this.props.loginFormData !== nextProps.loginFormData) {
-      // Ensuring browser's autofill user credentials get filled and their state persists in the redux store.
-      const nextState = {
-        emailOrUsername: nextProps.loginFormData.emailOrUsername || this.state.emailOrUsername,
-        password: nextProps.loginFormData.password || this.state.password,
-      };
-      this.setState({
-        ...nextProps.loginFormData,
-        ...nextState,
-      });
-      return false;
+  useEffect(() => {
+    if (loginErrorCode) {
+      setErrorCode(prevState => ({ type: loginErrorCode, count: prevState.count + 1 }));
     }
-    return true;
-  }
+  }, [loginErrorCode]);
 
-  componentWillUnmount() {
-    if (this.props.resetPassword) {
-      this.props.loginRemovePasswordResetBanner();
+  useEffect(() => {
+    if (loginResult.success) {
+      // TODO: Do we still need this cookie?
```

Comment:
I don't think we are running any surveys! Maybe we should tag Wajeeha here.

Reply:
Its on my agenda to discuss it with Wajeeha during standup.